components/proxy/src/test/java/com/hotels/styx/admin/handlers/ServiceProviderHandlerTest.java

```diff
@@ -0,0 +1,168 @@
+        assertThat(actualProviders.size(), equalTo(store.entrySet().size()));
+        for (StyxObjectDefinition actual :
+                actualProviders) {
+            Optional<ProviderObjectRecord> rec = store.get(actual.name());
+            assertTrue(rec.isPresent());
+            validateProvider(actual, rec.get());
+        }
+    }
+
+    @Test
+    public void returnsNoContentStatusWhenNoProvidersAvailable() {
+        StyxObjectStore<ProviderObjectRecord> empty = new StyxObjectStore<>();
+        ServiceProviderHandler handler = new ServiceProviderHandler(empty);
+        HttpRequest request = HttpRequest.get("/admin/service/providers").build();
+        HttpResponse response = Mono.from(handler.handle(request, HttpInterceptorContext.create())).block();
+
+        assertThat(response.status(), equalTo(HttpResponseStatus.NO_CONTENT));
+        assertThat(response.contentLength().get(), equalTo(0L));
+    }
+
+    @Test
+    public void returnsNamedProvider() throws IOException {
+        StyxObjectStore<ProviderObjectRecord> store = createTestStore();
+        ServiceProviderHandler handler = new ServiceProviderHandler(store);
+        HttpRequest request = HttpRequest.get("/admin/service/provider/object2").build();
+        HttpResponse response = Mono.from(handler.handle(request, HttpInterceptorContext.create())).block();
+
+        assertThat(response.status(), equalTo(HttpResponseStatus.OK));
+
+        StyxObjectDefinition actualProvider = deserialiseProvider(response.bodyAs(StandardCharsets.UTF_8));
+        assertThat(actualProvider.name(), equalTo("object2"));
+        assertThat(actualProvider, notNullValue());
+        validateProvider(actualProvider, store.get("object2").get());
+    }
+
+    @Test
+    public void returnsNotFoundStatusWhenNamedProviderNotFound() throws IOException {
+        StyxObjectStore<ProviderObjectRecord> store = createTestStore();
+        ServiceProviderHandler handler = new ServiceProviderHandler(store);
+        HttpRequest request = HttpRequest.get("/admin/service/provider/nonexistent").build();
+        HttpResponse response = Mono.from(handler.handle(request, HttpInterceptorContext.create())).block();
+
+        assertThat(response.status(), equalTo(HttpResponseStatus.NOT_FOUND));
+    }
+
+    @Test
+    public void returnsNotFoundStatusWithNonHandledUrl() {
+        StyxObjectStore<ProviderObjectRecord> empty = new StyxObjectStore<>();
+        ServiceProviderHandler handler = new ServiceProviderHandler(empty);
+        HttpRequest request = HttpRequest.get("/not/my/url").build();
+        HttpResponse response = Mono.from(handler.handle(request, HttpInterceptorContext.create())).block();
+
+        assertThat(response.status(), equalTo(HttpResponseStatus.NOT_FOUND));
+    }
+
+    private StyxObjectStore<ProviderObjectRecord> createTestStore() throws IOException {
+        ObjectMapper mapper =new ObjectMapper();
```

Comment:
I don't know if this is going to pass stylistic checks in checkstyle or sonar....  Add a whitespace before new?

Reply:
Even if it does, it's still ugly. Thx.